### PR TITLE
refactor(merkle_tree): general purpose nested list

### PR DESCRIPTION
**Pull request by 0xNineteen**
- uses a new general purposed nested list struct which can be used with any `[][]T` type

### CODE REVIEW

File: src/utils/merkle_tree.zig [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+                    const index_in_nested = index - search_index;
+                    return &slice[index_in_nested];
+                } else {
+                    search_index += slice.len;
+                }
+            }
+            unreachable;
+        }
+
+        pub fn getSlice(self: *const Self, index: u64, length: u64) []T {
```
Comment (InKryption) [inline src/utils/merkle_tree.zig:25]: What exactly is the point of this function? I don't see it used anywhere so it's hard to judge what exactly the point is; there's no API for getting the length of the slice that the element at `index` would be in, which makes me question how exactly the `length` should be computed by the caller, or what the usage is in general.

**Review by InKryption — COMMENTED** [review]
Just one question, otherwise looks solid.

File: src/utils/merkle_tree.zig (hunk 1 above)
Comment (0xNineteen) [reply src/utils/merkle_tree.zig]: ah great catch - it was some old code i used previously but dont need anymore

d89bf4a

**Review by InKryption — APPROVED** [review]
lgtm

**Pull request merged**